src/debugger/adapter/DebugAdapter.js

```diff
@@ -50,16 +50,20 @@ class PrepackDebugSession extends DebugSession {
   }
 
   _registerMessageCallbacks() {
+    this._ensureAdapterChannelCreated("registerMessageCallbacks");
+    invariant(this._adapterChannel !== undefined, "Adapter Channel used before it was created, in debugger.");
     this._adapterChannel.registerChannelEvent(DebugMessage.STOPPED_RESPONSE, (response: DebuggerResponse) => {
       let result = response.result;
       invariant(result.kind === "stopped");
       let message = `${result.reason}: ${result.filePath} ${result.line}:${result.column}`;
-      // Append message if there exists one (for PP errors)
+      // Append message if there exists one (for Prepack errors)
       if (result.message !== undefined) {
         message += `. ${result.message}`;
       }
       this.sendEvent(new StoppedEvent(message, DebuggerConstants.PREPACK_THREAD_ID));
     });
+    this._ensureAdapterChannelCreated("registerMessageCallbacks");
+    invariant(this._adapterChannel !== undefined, "Adapter Channel used before it was created, in debugger.");
```

Comment:
The same, no needed.

Reply:
You're right, the `_ensure...` call is unnecessary but the invariant needs to be there to satisfy flow (I believe because the previous call to `this._adapterChannel.registerChannelEvent` messes with its "knowledge".

Reply:
You can store `this._adapterChannel ` in a local variable so that later use of this local variable does not need invariant().

Reply:
I see that does satisfy flow, but it seems like a very unintuitive way of writing code -- I've never seen a global var. re-assigned to a local for use within a function (unless the goal was some copy+mutate operation). 

While two invariants is slightly more verbose and ugly, it would be clearer that flow was the reason, instead of some potentially unseen intention.

Reply:
I do not know what is `unintuitive`. Make a local copy is a way to explicitly tell type system that my state won't be changed by other external code. That's why invariant is unnecessary because you are helping type system to do the analysis. If your method got bigger and bigger, with 10+ using of `this._adapterChannel`, adding "invariant" 10+ times is not only annoying but also clutters the main logic of the code.

Reply:
I see. I initially thought it was the act of calling a method on an object that triggered flow -- protecting against external code modifying the object makes more sense now.